mas/src/main/java/com/ca/mas/foundation/JwtSignRequest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2016 CA. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.mas.foundation;
+
+import com.ca.mas.core.conf.ConfigurationManager;
+import com.ca.mas.core.conf.Server;
+import com.ca.mas.core.http.ContentType;
+import com.ca.mas.core.http.MAGRequest;
+import com.ca.mas.core.http.MAGRequestBody;
+import com.ca.mas.core.request.internal.MAGRequestProxy;
+import com.ca.mas.core.store.StorageProvider;
+import com.ca.mas.core.store.TokenManager;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.jwt.util.DateUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.util.Date;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+class JwtSignRequest extends MAGRequestProxy {
+
+    JwtSignRequest(MAGRequest request) {
+        this.request = request;
+    }
+
+    @Override
+    public MAGRequestBody getBody() {
+        MAGRequestBody body = super.getBody();
+        if (body != null) {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            try {
+                body.write(baos);
+                byte[] data = baos.toByteArray();
+                JWSSigner signer = new RSASSASigner(getPrivateKey());
+                JWTClaimsSet.Builder claimBuilder = new JWTClaimsSet.Builder();
+                TokenManager tokenManager = StorageProvider.getInstance().getTokenManager();
+
+                // JWT claims
+                // iss
+                String magId = tokenManager.getMagIdentifier();
+                claimBuilder.issuer("device://" + magId + "/$");
+
+                // TODO: sub: username
```

Comment:
The JwtSignRequest has been removed, it has been replaced with com.ca.mas.foundation.MASRequestBody#jwtClaimsBody, com.ca.mas.foundation.MAS#sign(com.ca.mas.foundation.MASClaims) and 
com.ca.mas.foundation.MAS#sign(com.ca.mas.foundation.MASClaims, java.security.PrivateKey)